pilot/pkg/model/push_context.go

```diff
@@ -1605,7 +1605,11 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) error {
 		return err
 	}
 
-	sortConfigByCreationTime(envoyFilterConfigs)
+	if features.SortEnvoyFilterByName {
+		sort.SliceStable(envoyFilterConfigs, func(i, j int) bool { return envoyFilterConfigs[i].Name < envoyFilterConfigs[j].Name })
```

Comment:
this may have conflicts, we need to sort by name *and* creation time I think.

We had an outage a long time ago due to this exact sort type on GW

Reply:
Ah. Thanks. Will fix that.

Reply:
I added name and namespace to make it unique. Is it sufficient or still we need creation time sort?

Reply:
Fqn will avoid collisions, but it will be a completely different ordering. I am circling back to adding patch-priority to the api with default being “in the middle”. Then we can sort by priority first and creation time second. That way a user that wants to order two patches can use them without affecting everything else.

Reply:
Ok. Cool. So what should I do with this? Should revert to name and then creation timestamp sort?

Reply:
For now. I have ordered by name and creation timestamp.

Reply:
@mandarjog Can you PTAL and see any other changes are needed?